pylearn2/costs/tests/test_term_1_0_l1_penalty.py

```diff
@@ -0,0 +1,78 @@
+"""
+Test term_1_0_l1_penalty
+"""
+import numpy as np
+from pylearn2.datasets.dense_design_matrix import DenseDesignMatrix
+from pylearn2.models.mlp import MLP, Sigmoid
+from pylearn2.train import Train
+from pylearn2.training_algorithms.sgd import SGD, ExponentialDecay
+from pylearn2.termination_criteria import And, EpochCounter, MonitorBased
+from pylearn2.costs.cost import SumOfCosts
+from pylearn2.costs.mlp import Default, L1WeightDecay
+
+
+def create_dataset():
+    """
+    Create a fake dataset to initiate the training
+    """
+    x = np.array([[0, 0, 0, 1, 0, 1, 1, 0, 0, 1, 0, 0, 1, 0, 1, 0, 0, 1, 0, 0],
+                  [11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 11, 12, 13, 14, 15,
+                   16, 17, 18, 19, 20],
+                  [0.3, 0.1, 0.8, 0.1, 0.2, 0.6, 0.83, 0.45, 0.0, 0.67, 0.3,
+                   0.74, 0.8, 0.1, 0.2, 0.46, 0.83, 0.45, 0.0, 0.67]])
+
+    y = np.array([0, 0, 0, 1, 0, 1, 1, 0, 0, 1, 0, 0,
+                  1, 0, 1, 0, 0, 0, 0, 0]).reshape(20, 1)
+
+    x_train = x[:7, :]
+    x_valid = x[7:, :]
+    y_train = y[:7]
+    y_valid = y[7:]
+
+    return x_train, y_train, x_valid, y_valid
+
+
+def test_correctness():
+    """
+    Test that the cost function works with float64
+    """
+    x_train, y_train, x_valid, y_valid = create_dataset()
+
+    trainset = DenseDesignMatrix(X=np.array(x_train), y=y_train)
+    validset = DenseDesignMatrix(X=np.array(x_valid), y=y_valid)
+
+    n_inputs = trainset.X.shape[1]
+    n_outputs = 1
+    n_hidden = 10
+
+    hidden_istdev = 4 * (6 / (n_inputs + n_hidden)) ** 0.5
+    output_istdev = 4 * (6 / (n_hidden + n_outputs)) ** 0.5
```

Comment:
This will initialize the weights to 0 on Python 2, since the `/` is integer division.